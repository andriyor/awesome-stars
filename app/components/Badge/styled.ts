import { number } from "prop-types";
import styled from "styled-components";
import { numberToColor } from "../../helpers/numberToColor";

export const StyledBadge = styled.span`
    background-color: #3f3f3f;
    border-radius: 1rem;
    color: ${p => numberToColor(p.number)};
    font-family: Roboto, sans-serif;
    padding: 0.3125rem 0.4375rem;
`;

StyledBadge.propTypes = {
    number: number
};

StyledBadge.defaultProps = {
    number: 0
};
