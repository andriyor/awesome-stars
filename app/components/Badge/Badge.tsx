import * as React from "react";
import { FunctionComponent } from "react";
import { createGlobalStyle } from "styled-components";
import { StyledBadge, StyledStar } from "./styled";

interface BadgePropTypes {
    stargazerCount?: number;
}

const GlobalStyle = createGlobalStyle`
@import url('https://fonts.googleapis.com/css?family=Roboto');
`;

const numberFormat = new Intl.NumberFormat("en-US");

const Badge: FunctionComponent<BadgePropTypes> = (props) => {
    const { stargazerCount = 0 } = props;
    return (
        <React.Fragment>
            <GlobalStyle/>
            <StyledBadge stargazerCount={stargazerCount}>
                <StyledStar stargazerCount={stargazerCount}/>
                <span>{numberFormat.format(stargazerCount)}</span>
            </StyledBadge>
        </React.Fragment>
    );
};

export default Badge;
