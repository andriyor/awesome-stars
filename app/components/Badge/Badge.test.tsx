import * as React from "react";
import Badge from "./Badge";
import { shallow } from "enzyme";

test("Badge", () => {
    const badge = shallow(<Badge/>);
    expect(badge).toMatchSnapshot();
});

test("Badge with stargazer count", () => {
    const badge = shallow(<Badge stargazerCount={100}/>);
    expect(badge).toMatchSnapshot();
});
