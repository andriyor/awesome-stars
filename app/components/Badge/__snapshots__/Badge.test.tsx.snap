// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <styled.span
    stargazerCount={0}
  >
    <Styled(Star)
      stargazerCount={0}
    />
    <span>
      0
    </span>
  </styled.span>
</Fragment>
`;

exports[`Badge with stargazer count 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <styled.span
    stargazerCount={100}
  >
    <Styled(Star)
      stargazerCount={100}
    />
    <span>
      100
    </span>
  </styled.span>
</Fragment>
`;
